### PR TITLE
JWT ExpiresIn

### PR DESCRIPTION
**Pull request by macleysousa**
## JWT ExpiresIn
To configure the JWT expiration time add the parameter jwtExpiresIn in 'config/application.json'
```json
{
  "favicon": {
      "path": "favicon.ico",
      "maxAge": 86400000
  },
  "public": {
      "path": "./public",
      "maxAge": 60000
  },
  "jwtExpiresIn": "2h"
}
```
- jwtExpiresIn: `30d` time in days or `2h` time in hours

> Check if there is a parameter defined in '../../../config/application.json' for 'jwtExpiresIn'
if there is any value if it does not continue with the default value
 ```js
const { jwtExpiresIn } = require('../../../config/application.json');

const defaultJwtOptions = { expiresIn: jwtExpiresIn == null ? '30d' : jwtExpiresIn };
```

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/services/Jwt.js [hunk 1]
```diff
@@ -9,7 +9,24 @@
 const _ = require('lodash');
 const jwt = require('jsonwebtoken');
 
-const defaultJwtOptions = { expiresIn: '30d' };
+/*
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-users-permissions/services/Jwt.js:12]: Remove the example from here

**Comment by derrickmehaffy** [comment]
@macleysousa you may want to check this against the new config system coming in rc.0 as Strapi no longer defaults to using `.json` for config files

File: packages/strapi-plugin-users-permissions/services/Jwt.js (hunk 1 above)
Comment (macleysousa) [reply packages/strapi-plugin-users-permissions/services/Jwt.js]: example removed

**Comment by derrickmehaffy** [comment]
@alexandrebodin thoughts? I'm not sure how this best plays into the new config restructuring

**Comment by alexandrebodin** [comment]
Hello This PR will nott be acceptable as is. The config should not be loaded by requiring a file but by using strapi config loader. With the next config system it would be as simple as adding the config to a plugins.js config file. 

`./config/plugins.js`
```js
{
  "users-permissions": {
    "jwt": {
       "expiresIn": "2d"
     }
  }
}
```

This config will be merged into strapi.plugins.users-permissions.config directly so you could do 

`strapi.plugins.users-permissions.config.jwt.expiresIn`

**Comment by macleysousa** [comment]
@alexandrebodin do you have any prediction of when this feature will be launched? I don't know if this feature will already allow this, but it would be interesting to be able to set the jwt expiration time directly by the admin panel

**Comment by diogotcorreia** [comment]
Not sure if it's just me, but it looks like the `packages/strapi-plugin-users-permissions/services/Jwt.js` file was renamed to `Pacotes/strapi-plugin-users-permissões/Serviços/Jwt.js` in the latest commit (1e6c510) for this PR.

**Comment by macleysousa** [comment]
@diogotcorreia do not know how it happened but it was renamed

**Comment by alexandrebodin** [comment]
> @alexandrebodin do you have any prediction of when this feature will be launched? I don't know if this feature will already allow this, but it would be interesting to be able to set the jwt expiration time directly by the admin panel

What feature are you refering to ?

**Comment by derrickmehaffy** [comment]
@macleysousa you won't be able to change it inside the Admin as the value is stored in files (meaning it can't be changed in production) Currently I believe the only thing that is ever pushed into the Admin is the Content-Type builder (which is disabled in production/staging) but generally speaking if a config is file based it won't be added to the Admin. (Config options in the admin are typically stored in the database)

**Comment by macleysousa** [comment]
@alexandrebodin you said `With the next config system it would be as simple as adding the config to a plugins.js config file`
I would like to know in which version I could use this feature currently I am using the version `v3.0.0-beta.19.3`

**Comment by alexandrebodin** [comment]
This feature is available in the 3.0.0-rc.0 version and in the master branch of the repo

**Comment by macleysousa** [comment]
@alexandrebodin,  @derrickmehaffy,  @diogotcorreia 
>  thanks for listening

**Pull request closed without merging**